sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp

```diff
@@ -0,0 +1,432 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "../src/private/crypto/inc/crypto.hpp"
+#include "attestation_collateral.hpp"
+#include "azure/attestation/attestation_administration_client.hpp"
+#include "azure/identity/client_secret_credential.hpp"
+#include <azure/attestation/attestation_client_models.hpp>
+#include <azure/core/internal/cryptography/sha_hash.hpp>
+#include <azure/core/test/test_base.hpp>
+#include <cstdlib>
+#include <gtest/gtest.h>
+#include <string>
+#include <tuple>
+
+using namespace Azure::Security::Attestation;
+using namespace Azure::Security::Attestation::Models;
+using namespace Azure::Security::Attestation::_detail;
+
+namespace Azure { namespace Security { namespace Attestation { namespace Test {
+
+  enum class TestCaseType
+  {
+    GetPolicy,
+    ModifyPolicyUnsecured,
+    ModifyPolicySecured,
+    ModifyPolicyIsolated,
+  };
+
+  enum class ServiceInstanceType
+  {
+    Shared,
+    AAD,
+    Isolated
+  };
+
+  struct PolicyTestParam
+  {
+    TestCaseType TestType;
+    ServiceInstanceType InstanceType;
+    AttestationType TeeType;
+  };
+
+  class PolicyTests : public Azure::Core::Test::TestBase,
+                      public testing::WithParamInterface<PolicyTestParam> {
+  private:
+  protected:
+    std::shared_ptr<Azure::Core::Credentials::TokenCredential> m_credential;
+    std::string m_endpoint;
+
+    // Create
+    virtual void SetUp() override
+    {
+      Azure::Core::Test::TestBase::SetUpTestBase(AZURE_TEST_RECORDING_DIR);
+
+      if (GetParam().InstanceType == ServiceInstanceType::Shared)
```

Comment:
I would save `GetParam().InstanceType` into a const var and then do the if/else if